backend/routes/rankings.js

```diff
@@ -1,52 +1,81 @@
 import express from "express";
 import { Follow } from "../models/follow.js";
 import { Stock } from "../models/stock.js";
+import { Like } from "../models/like.js";
+import { Dislike } from "../models/dislike.js";
+import { compareStocksByPoints } from "../utils.js";
+
+const FOLLOW_POINTS = 25;
+const LIKE_POINTS = 20;
+const DISLIKE_POINTS = -20;
 
 const router = express.Router();
 
 router.get("/ranking", async (req, res) => {
     const user = req.session.user;
 
     try {
-        const follows = await Follow.findAll({
-            where: { UserId: user.id },
-            include: [{ model: Stock }]
-        });
-
-        const stocksYouFollow = follows.length === 0 ? [] : follows.map((follow) => follow.Stock);
-
-        const stocksInDatabase = await Stock.findAll();
+        const [stocksInDatabase, follows, likes, dislikes] = await Promise.all([
+            Stock.findAll(),
+            Follow.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Like.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Dislike.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            })
+        ]);
 
         if (stocksInDatabase.length < 10) {
             res.status(422).json({
                 error: "You need at least 10 stocks in the database to access ranking."
             });
         }
 
-        const randomStocksNeeded = 10 - stocksYouFollow.length;
+        const stocksYouFollow = follows.map((follow) => follow.Stock);
+        const stocksYouLike = likes.map((like) => like.Stock);
+        const stocksYouDislike = dislikes.map((dislike) => dislike.Stock);
 
-        const stocksRanking = [...stocksYouFollow];
+        const stocksRanking = [];
 
-        for (let i = 0; i < randomStocksNeeded; i++) {
-            let potentialStock;
-            let alreadyInRanking;
+        stocksInDatabase.forEach((currentStock) => {
+            let points = 0;
 
-            do {
-                const randomIndex = Math.floor(
-                    Math.random() * stocksInDatabase.length
-                );
-                potentialStock = stocksInDatabase[randomIndex];
+            const selectedStockYouFollow = stocksYouFollow.find((stock) => stock.ticker === currentStock.ticker);
```

Comment:
Something to think about in design is scale and picking the right data structure. In small examples, what you have here works fine and you won't see major performance slowdowns. 

But say `stocksYouFollow` is a very large list of size N -- then for each stock in the DB (say DB size M), you're doing a lookup to find the stock whose ticker matches. This easily becomes an expensive operation (`O(MN)`, here's a [quick ref](https://www.freecodecamp.org/news/big-o-notation-why-it-matters-and-why-it-doesnt-1674cfa8a23c/) on Big-O)

`stocksYouFollow` is currently a list, but using something like a Set or dictionary with an index on the ticker could be better for faster lookup operations.

same idea follows for other examples below when looking up from `stocksYouLike` and `stocksYouDislike`